PYIC-1065: Lambda and API gateway logs to Splunk

Wynndow opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Add CloudFormation resources to start sending our lambda application
logs and API gateway access logs to Splunk.

### Why did it change

For the API gateway logs I've reused the JSON format we have defined for
the core-front Fargate API gateway logs, for consistency. The only
difference is the addition of `responseLatency` which could be helpful
when looking at slow lambdas.

For the lambdas I've explicitly declared a log group resource even
though lambda creates one for you. This is because the lambda created
log groups will store messages indefinitely by default. You can change
this by declaring the log group yourself. As long as the name is the
same as the log group lambda creates and writes to, the function should
write to this new log group. It will be interesting to see if these log
groups deploy cleanly, considering they already exist...

The subscription filters are taking the logs from the newly created log
groups and sending them to Cyber's Centralised Security Logging Service
(CSLS) Kinesis data stream. This is the recommended way of getting our
logs to Splunk. See here for more details:
https://github.com/alphagov/centralised-security-logging-service

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-1065](https://govukverify.atlassian.net/browse/PYIC-1065)

Wynndow committed:
PYIC-1065: Lambda and API gateway logs to Splunk

Add CloudFormation resources to start sending our lambda application
logs and API gateway access logs to Splunk.

For the API gateway logs I've reused the JSON format we have defined for
the core-front Fargate API gateway logs, for consistency. The only
difference is the addition of `responseLatency` which could be helpful
when looking at slow lambdas.

For the lambdas I've explicitly declared a log group resource even
though lambda creates one for you. This is because the lambda created
log groups will store messages indefinitely by default. You can change
this by declaring the log group yourself. As long as the name is the
same as the log group lambda creates and writes to, the function should
write to this new log group. It will be interesting to see if these log
groups deploy cleanly, considering they already exist...

The subscription filters are taking the logs from the newly created log
groups and sending them to Cyber's Centralised Security Logging Service
(CSLS) Kinesis data stream. This is the recommended way of getting our
logs to Splunk. See here for more details:
https://github.com/alphagov/centralised-security-logging-service

danworth commented on deploy/template.yaml:
```diff
@@ -26,6 +26,38 @@ Resources:
+    Properties:
+      DestinationArn: "arn:aws:logs:eu-west-2:885513274347:destination:csls_cw_logs_destination_prod"
+      FilterPattern: ""
+      LogGroupName: !Select # Select the string at the 6th index of...
```
You can replace this with simply `!Ref IPVCoreInternalAPILogGroup` which returns the log group name, here and in the similar places below. Usually `!Ref <logical resource id>` gives you the name of the resource, useful for dynamodb tables etc.

Wynndow replied:
Oh mate. I was looking at the docs and completely missed the Ref part, focussed on the Attributes section. This is going to make things a lot easier...

Wynndow replied:
Fixed up

Wynndow committed:
PYIC-1065: Remove whitespace from access logs format

The whitespace is kept during processing and turns up in the logs in
Splunk. It looks confusing.

vam-gds reviewed (approved):
lgtm
